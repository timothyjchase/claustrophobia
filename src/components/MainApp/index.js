import React from 'react'
import PropTypes from 'prop-types'
import { Container } from 'semantic-ui-react'
import GameSummaryContainer from '../../containers/GameSummaryContainer'
import EventMessageContainer from '../../containers/EventMessageContainer'
import Welcome from '../Welcome'
import PhaseHeader from '../PhaseHeader'
import PhaseBody from '../PhaseBody'
import MainMenu from '../MainMenu'

const MainApp = ({
  currentState,
  scenarioName,
  startGame,
  resetGame,
  undoLastChange,
}) => (
  <Container>
    <div
      style={{
        maxWidth: '352px',
        margin: '0 auto',
        background: 'white',
        marginTop: '10px',
      }}
    >
      <img
        src="images/title.png"
        alt="Claustrophobia"
        style={{ width: '352px', height: '128px' }}
      />

      <MainMenu
        style={{
          position: 'absolute',
          zIndex: '1000',
          top: '110px',
          width: '352px',
        }}
        currentState={currentState}
        scenarioName={scenarioName}
        resetGame={resetGame}
        undoLastChange={undoLastChange}
      />
      {!!scenarioName && (
        <div
          style={{
            position: 'absolute',
            left: '50%',
            top: '115px',
            zIndex: '1000',
          }}
        >
          <div
            style={{
              position: 'relative',
              left: '-50%',
            }}
          >
            <strong>{scenarioName}</strong>
          </div>
        </div>
      )}

      <div style={{ padding: '5px 10px 10px 10px' }}>
        {!scenarioName && <Welcome startGame={startGame} />}
        {!!scenarioName && (
          <div>
            <PhaseHeader phase={currentState.phase} />
            <EventMessageContainer />
            <PhaseBody phase={currentState.phase} />
            <GameSummaryContainer />
          </div>
        )}
      </div>
    </div>
  </Container>
)

MainApp.propTypes = {
  currentState: PropTypes.object.isRequired,
  scenarioName: PropTypes.string, // eslint-disable-line react/require-default-props
  startGame: PropTypes.func.isRequired,
  resetGame: PropTypes.func.isRequired,
  undoLastChange: PropTypes.func.isRequired,
}

export default MainApp
