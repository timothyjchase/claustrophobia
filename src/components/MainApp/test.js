import React from 'react'
import MainApp from './index'

describe('<MainApp />', () => {
  const commonProps = {
    currentState: { phase: 'test phase' },
    startGame: jest.fn(),
    resetGame: jest.fn(),
    undoLastChange: jest.fn(),
  }
  describe('with scenarioName', () => {
    const component = shallow(
      <MainApp {...commonProps} scenarioName="Test scenario" />,
    )
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with no scenarioName', () => {
    const component = shallow(<MainApp {...commonProps} />)
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
