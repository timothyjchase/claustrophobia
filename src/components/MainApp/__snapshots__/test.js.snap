// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainApp /> with no scenarioName should match snapshot 1`] = `
<Container>
  <div
    style={
      Object {
        "background": "white",
        "margin": "0 auto",
        "marginTop": "10px",
        "maxWidth": "352px",
      }
    }
  >
    <img
      alt="Claustrophobia"
      src="images/title.png"
      style={
        Object {
          "height": "128px",
          "width": "352px",
        }
      }
    />
    <MainMenu
      currentState={
        Object {
          "phase": "test phase",
        }
      }
      resetGame={[MockFunction]}
      style={
        Object {
          "position": "absolute",
          "top": "110px",
          "width": "352px",
          "zIndex": "1000",
        }
      }
      undoLastChange={[MockFunction]}
    />
    <div
      style={
        Object {
          "padding": "5px 10px 10px 10px",
        }
      }
    >
      <Welcome
        startGame={[MockFunction]}
      />
    </div>
  </div>
</Container>
`;

exports[`<MainApp /> with scenarioName should match snapshot 1`] = `
<Container>
  <div
    style={
      Object {
        "background": "white",
        "margin": "0 auto",
        "marginTop": "10px",
        "maxWidth": "352px",
      }
    }
  >
    <img
      alt="Claustrophobia"
      src="images/title.png"
      style={
        Object {
          "height": "128px",
          "width": "352px",
        }
      }
    />
    <MainMenu
      currentState={
        Object {
          "phase": "test phase",
        }
      }
      resetGame={[MockFunction]}
      scenarioName="Test scenario"
      style={
        Object {
          "position": "absolute",
          "top": "110px",
          "width": "352px",
          "zIndex": "1000",
        }
      }
      undoLastChange={[MockFunction]}
    />
    <div
      style={
        Object {
          "left": "50%",
          "position": "absolute",
          "top": "115px",
          "zIndex": "1000",
        }
      }
    >
      <div
        style={
          Object {
            "left": "-50%",
            "position": "relative",
          }
        }
      >
        <strong>
          Test scenario
        </strong>
      </div>
    </div>
    <div
      style={
        Object {
          "padding": "5px 10px 10px 10px",
        }
      }
    >
      <div>
        <PhaseHeader
          phase="test phase"
        />
        <Connect(EventMessageContainer) />
        <PhaseBody
          phase="test phase"
        />
        <Connect(GameSummaryContainer) />
      </div>
    </div>
  </div>
</Container>
`;
